File: docs-examples/example-java/src/test/java/io/micronaut/rabbitmq/docs/rpc/RpcUppercaseSpec.java
package io.micronaut.rabbitmq.docs.rpc;

import io.micronaut.context.annotation.Property;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import io.micronaut.test.support.TestPropertyProvider;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInstance;
import reactor.core.publisher.Mono;

import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertEquals;
import io.micronaut.rabbitmq.testcontainers.RabbitMQ;

@MicronautTest
@Property(name = "spec.name", value = "RpcUppercaseSpec")
@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class RpcUppercaseSpec implements TestPropertyProvider {
    @Override
    public Map<String, String> getProperties() {
        return RabbitMQ.getProperties();
    }

    @Test
    void testProductClientAndListener(ProductClient productClient) {
// tag::producer[]
assertEquals("RPC", productClient.send("rpc"));
assertEquals("HELLO", Mono.from(productClient.sendReactive("hello")).block());
// end::producer[]
    }
}

File: docs-examples/example-python/src/test/python/micronaut/rabbitmq/docs/rpc/RpcUppercaseSpec.py
from typing import Annotated

from jakarta.inject import Inject
from micronaut.context.annotation import Property
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test
from reactor.core.publisher import Mono

from .ProductClient import ProductClient


@MicronautTest(environments=["rabbitmq"])
@Property(name="spec.name", value="RpcUppercaseSpec")
class RpcUppercaseSpec:
    product_client: Annotated[ProductClient, Inject]

    @Test
    def test_product_client_and_listener(self) -> None:
# tag::producer[]
        assert self.product_client.send("rpc") == "RPC"
        assert Mono.from_(self.product_client.send_reactive("hello")).block() == "HELLO"
# end::producer[]
